dist/index.d.ts

```diff
@@ -3880,68 +3883,92 @@ interface RoomPosition {
      * @param opts An object containing pathfinding options (see Room.findPath), or one of the following: filter, algorithm
      * @returns An instance of a RoomObject.
      */
-    findClosestByPath<K extends FindConstant, S extends FindTypes[K]>(
+    findClosestByPath<K extends FindConstant, T extends FindTypes[K], S extends T>(
         type: K,
-        opts?: FindPathOpts & Partial<FilterOptions<K, S>> & { algorithm?: FindClosestByPathAlgorithm },
+        opts?: FindPathOpts & Partial<PredicateFilterOptions<T, S>> & { algorithm?: FindClosestByPathAlgorithm },
     ): S | null;
-    findClosestByPath<S extends AnyStructure>(
+    findClosestByPath<K extends FindConstant, T extends FindTypes[K] = FindTypes[K]>(
+        type: K,
+        opts?: FindPathOpts & Partial<FilterOptions<FindTypes[K]>> & { algorithm?: FindClosestByPathAlgorithm },
+    ): T | null;
+    findClosestByPath<T extends AnyStructure>(
         type: FIND_STRUCTURES | FIND_MY_STRUCTURES | FIND_HOSTILE_STRUCTURES,
-        opts?: FindPathOpts & Partial<FilterOptions<FIND_STRUCTURES, S>> & { algorithm?: FindClosestByPathAlgorithm },
-    ): S | null;
+        opts?: FindPathOpts & Partial<FilterOptions<AnyStructure>> & { algorithm?: FindClosestByPathAlgorithm },
```

Comment:
Maybe we can infer this `AnyStructure` from the first argument, but another type parameter would be introduced.

Reply:
> Is this about the declaration starting on line 3984? You want to infer the type of application of `findClosestByPath` from the type of first argument of function defined in `filter`? Wouldn't that be difficult due to `FilterOption.filter` being of a union type? Isn't it already covered by the version from line 3886?

I mean inferring the `AnyStructure` from the given `FIND_(MY)_STRUCTURE`. 
TS throws an error in the case below.
```
{
    let towers = creep.pos.findClosestByPath<StructureTower>(FIND_MY_STRUCTURES, {
        filter: (s) => s?.my // ERROR: `my` is not exist on `AnyStructure`
    });
}
```

Reply:
You could do `F extends FIND_STRUCTURES | FIND_MY_STRUCTURES | FIND_HOSTILE_STRUCTURES` and then use `FindTypes[F]` to get the structure type, I think. Or at least a similar idea to that with a map.

Reply:
Let's keep it as is now.